File: 36-valid-sudoku/valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        unordered_set<string> m;
        int rows = board.size();
        int cols = board[0].size();
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (board[i][j] != '.') {
                    string row = "row" + to_string(i) + to_string(board[i][j]);
                    string col = "col" + to_string(j) + to_string(board[i][j]);
                    string box = "box" +
                                 to_string(i / 3) + to_string(j / 3) +
                                 to_string(board[i][j]);
                    if (m.count(row) || m.count(col) || m.count(box)) {
                        return false;
                    } else {
                        m.insert(row);
                        m.insert(col);
                        m.insert(box);
                    }
                }
            }
        }
        return true;
    }
};

File: 36-valid-sudoku/valid-sudoku.js
var isValidSudoku = function (board) {
    let row = new Set();
    let col = new Set();
    let box = new Set();

    const rows = board.length;
    const cols = board[0].length;

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < cols; j++) {
            if (board[i][j] !== ".") {
                const r = `row ${i} value ${board[i][j]}`;
                const c = `col ${j} value ${board[i][j]}`;
                const b = `box ${Math.floor(i / 3)}-${Math.floor(j / 3)} value ${board[i][j]}`;

                if (row.has(r) || col.has(c) || box.has(b)) {
                    return false;
                }

                row.add(r);
                col.add(c);
                box.add(b);
            }
        }
    }
    return true;
};